### PR TITLE
client-go: promote exec plugin support to beta

### PR DESCRIPTION
**Pull request by ericchiang**
/sig auth
/kind feature

Adds a "v1beta1" API group for client authentication. Because of a lack of usage with the input parameters, these have been dropped for the beta. Would like to re-introduce them as users feel they require them.

updates https://github.com/kubernetes/kubernetes/issues/61796

```release-note
client-go: credential exec plugins have been promoted to beta
```

### CODE REVIEW

**Comment by ericchiang** [comment]
Going to let https://github.com/kubernetes/kubernetes/pull/61803 merge first.

**Comment by ericchiang** [comment]
cc @kubernetes/sig-auth-pr-reviews

File: staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec_test.go [hunk 1]
```diff
@@ -420,6 +486,13 @@ func TestRefreshCreds(t *testing.T) {
 				t.Errorf("expected expiry %v got %v", test.wantExpiry, a.exp)
 			}
 
+			if test.wantInput == "" {
+				if got := strings.TrimSpace(stderr.String()); got != "" {
+					t.Errorf("expectd no input parameters, got %q", got)
```
Comment (php-coder) [inline staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec_test.go]: s/expectd/expected/

Comment (ericchiang) [reply staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec_test.go]: thanks! updated

File: staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1beta1/types.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	Token string `json:"token,omitempty"`
+	// PEM-encoded client TLS certificates (including intermediates, if any).
+	ClientCertificateData string `json:"clientCertificateData,omitempty"`
+	// PEM-encoded private key for the above certificate.
+	ClientKeyData string `json:"clientKeyData,omitempty"`
+}
+
+// Response defines metadata about a failed request, including HTTP status code and
+// response headers.
+type Response struct {
```
Comment (awly) [inline staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1beta1/types.go]: Can this be removed?

Comment (ericchiang) [reply staging/src/k8s.io/client-go/pkg/apis/clientauthentication/v1beta1/types.go]: Yep. thanks for pointing that out

**Comment by ericchiang** [comment]
/assign @thockin 

For final approval

**Comment by thockin** [comment]
I have no context on this API, so it's hard for me to review.  Given that it is mostly empty, I guess it's not a big deal yet :)

/approve

**Pull request merged**